env0/utils.go

```diff
@@ -134,6 +119,54 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 	return nil
 }
 
+func readResourceDataSliceStructHelper(field reflect.Value, resource interface{}) error {
+	val := field.Elem()
+	m := resource.(map[string]interface{})
+
+	for i := 0; i < val.NumField(); i++ {
+		fieldName, skip := getFieldName(val.Type().Field(i))
+		if skip {
+			continue
+		}
+
+		fieldValue, ok := m[fieldName]
+		if !ok {
+			continue
+		}
+
+		field := val.Field(i)
+		field.Set(reflect.ValueOf(fieldValue))
+	}
+
+	return nil
+}
+
+// Extracts a list of values from the resourcedata, and writes it to the struct field.
+func readResourceDataSlice(field reflect.Value, resources []interface{}) error {
```

Comment:
It's hard to read/understand reflection code.
Please ping me if anything is unclear.

Reply:
This is already tested in utils_test.
For slice of structs and slice of strings.